src/Http/Routing/src/Constraints/AlphaRouteConstraint.cs

```diff
@@ -1,17 +1,22 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Text.RegularExpressions;
+
 namespace Microsoft.AspNetCore.Routing.Constraints;
 
 /// <summary>
 /// Constrains a route parameter to contain only lowercase or uppercase letters A through Z in the English alphabet.
 /// </summary>
-public class AlphaRouteConstraint : RegexRouteConstraint
+public partial class AlphaRouteConstraint : RegexRouteConstraint
 {
     /// <summary>
     /// Initializes a new instance of the <see cref="AlphaRouteConstraint" /> class.
     /// </summary>
-    public AlphaRouteConstraint() : base(@"^[a-z]*$")
+    public AlphaRouteConstraint() : base(GetAlphaRouteRegex())
     {
     }
+
+    [GeneratedRegex(@"^[A-Za-z]*$", RegexOptions.CultureInvariant)]
```

Comment:
```suggestion
    [GeneratedRegex(@"^[A-Za-z]*$")]
```
Without IgnoreCase or `(?i)` in the pattern, CultureInvariant is a nop.

Reply:
Technically this is missing the 10_000 milliseconds timeout that's currently being passed to the Regex ctor in the existing code.  In practice, it's not relevant to this usage, as we don't check for the timeout in the middle of the set loop (which will end up being implemented as a vectorized IndexOfAny in the very near future), and as the pattern is anchored, we won't be doing the search more than once.  Just calling it out as an FYI.

Reply:
> Technically this is missing the 10_000 milliseconds timeout that's currently being passed to the Regex ctor in the existing code.

Are you referring to existing code in the runtime here? In what scenarios is it required to override the default timeout? 🤔

Reply:
I'm referring to the ASP.NET code passing in 10s as a timeout:
https://github.com/dotnet/aspnetcore/blob/9e2c389ab7f31d45e4e805edba34341ff859749b/src/Http/Routing/src/Constraints/RegexRouteConstraint.cs#L47

Reply:
> In what scenarios is it required to override the default timeout? 🤔

The default timeout is infinite. You'd override it if you were concerned about runaway execution due to a pattern encountering catastrophic backtracking.  The pattern in this PR will result in time linear in the length of the input.